src/sparseml/pytorch/sparsification/quantization/helpers.py

```diff
@@ -69,6 +92,177 @@
 )
 
 
+@dataclass
+class QConfigProperties:
+    """
+    Dataclass that stores properties needed to define qconfig objects.
+    Default values set here.
+
+    :param symmetric_activations: if True, activations will have a symmetric
+        quantization range with a pre-specified zero point
+        (0 if activation_dtype=torch.qint8, 128 if activation_dtype=torch.quint8).
+        Default is False.
+    :param symmetric_weights: if True, weights will have a symmetric
+        quantization range with a pre-specified zero point
+        (0 if weight_dtype=torch.qint8, 128 if weight_dtype=torch.quint8).
+        Default is True.
+    :param reduce_range: if True, the quantization range will be reduced by one bit.
+        This may prevent overflow issues with model execution on certain hardware.
+        Default is False.
+    :param activation_qconfig_kwargs: Additional kwargs for quantization of
+        activations.
+    :param weight_qconfig_kwargs: Additional kwargs for quantization of
+        weights.
+    :param activation_dtype: quantized activation data type.
+        Default is torch.quint8.
+    :param weight_dtype: quantized weights data type.
+        Default is torch.qint8.
+    :param activation_bits: number of bits for activations. Default is 8.
+    :param weight_bits: number of bits for weights. Default is 8.
+    """
+
+    _symmetric_activations: Optional[bool] = None
+    _symmetric_weights: Optional[bool] = None
+    reduce_range: bool = False
+    activation_dtype: torch.dtype = torch.quint8
+    weight_dtype: torch.dtype = torch.qint8
+    activation_bits: int = 8
+    weight_bits: int = 8
+    activation_qconfig_kwargs: Dict[str, Any] = field(default_factory=dict)
+    weight_qconfig_kwargs: Dict[str, Any] = field(default_factory=dict)
+
+    @property
+    def symmetric_activations(self) -> bool:
+        if self._symmetric_activations:
+            return self._symmetric_activations
+        else:
+            return False
+
+    @symmetric_activations.setter
+    def symmetric_activations(self, value: bool):
+        if self._symmetric_activations is None:
+            self._symmetric_activations = value
+
+    @property
+    def symmetric_weights(self) -> bool:
+        if self._symmetric_weights:
+            return self._symmetric_weights
+        else:
+            return True
+
+    @symmetric_weights.setter
+    def symmetric_weights(self, value: bool):
+        if self._symmetric_weights is None:
+            self._symmetric_weights = value
+
+
+class BNWrapper(Module):
```

Comment:
if we don't plan on using this outside of this file, let's rename to `_BNWrapper` and move to bottom of file. Also in the docstring, should specify that this is for when the BN is not fused

Reply:
Done